analyzer/src/test/kotlin/managers/GoModTest.kt

```diff
@@ -70,4 +73,59 @@ class GoModTest : WordSpec({
             id.toVcsInfo().revision shouldBe "v43.3.0"
         }
     }
+
+    "parseWhyOutput" should {
+        "detect packages that are used" {
+            val output = listOf(
+                "# cloud.google.com/go",
+                "foo",
+                "# github.com/fatih/color",
+                "bar",
+                "# github.com/hashicorp/go-multierror",
+                "baz"
+            ).asSequence()
+
+            parseWhyOutput(output) should containExactlyInAnyOrder(
+                "cloud.google.com/go",
+                "github.com/fatih/color",
+                "github.com/hashicorp/go-multierror"
+            )
+        }
+
+        "handle a missing current package" {
+            val output = listOf("foo").asSequence()
+
+            parseWhyOutput(output) should beEmpty()
+        }
+
+        "skip unused packages" {
+            val output = listOf(
+                "# cloud.google.com/go",
```

Comment:
My proposal to use `String` instead of `Sequence` would lead to multi line strings in this test which is IMO a bit nicer to read and write.